[RHACS] ROX-11713: Remove stackrox.io repo for images (#47625)
openshift/openshift-docs: modules/central-services-private-config.adoc

[id="central-services-private-configuration-file-image-pull-secrets_{context}"]
== Image pull secrets

> The credentials required for pulling images from the registry.

Reviewer: ```suggestion
The credentials that are required for pulling images from the registry depend on the following factors:
```

If you are using a custom registry, you must specify these parameters:

* `imagePullSecrets.username`
* `imagePullSecrets.password`
* `image.registry` 

If you do not use a username and password to log in to the custom registry, you must specify one of the following parameters:

* `imagePullSecrets.allowNone`
* `imagePullSecrets.useExisting`
* `imagePullSecrets.useFromDefaultServiceAccount`

|===
| Parameter | Description

| `imagePullSecrets.username`

> | The username for the registry.

Reviewer: ```suggestion
| The username of the account that is used to log into the registry.
```

| `imagePullSecrets.password`

> | The password (for the selected username) of the registry.

Reviewer: ```suggestion
| The password of the account that is used to log in to the registry.
```

== Image pull secrets
The credentials required for pulling images from your registry.

If you are using a custom registry, you must specify these parameters:

* `imagePullSecrets.username`
* `imagePullSecrets.password`
* `image.registry` 

> If you do not use a username and password to log in to your custom registry, you must specify one of the following parameters:

Reviewer 2: Suggesting to rephrase it a bit, optionally. Primarily thinking about enterprises where the person installing ACS is not controlling the image registry therefore "your custom registry" does not apply.

```suggestion
If you do not use a username and password to log in to the custom image registry, you must specify one of the following parameters:
```

[NOTE]
====

> * If you are using a custom registry, you must also specify the `image.registry` parameter.

Reviewer 2: I think this should say something like
```suggestion
* If you are using a custom registry, you must specify the `imagePullSecrets.username`, `imagePullSecrets.password` and `image.registry` parameters.
```

Moreover, I suspect that only on OpenShift you don't need `imagePullSecrets.username` and `imagePullSecrets.password` for default `registry.redhat.io` because OpenShift authenticates globally to it.
On non-OpenShift Kubernetes `imagePullSecrets.username` and `imagePullSecrets.password` may be needed for `registry.redhat.io`. However, this needs to be tested as I'm not fully confident.

Reviewer 2: Well, I suggest to a) apply you removal to `If you are using the `stackrox.io` registry` line, b) apply my changes to `If you are using a custom registry` line. We will not be wrong this way.

The point about necessity of `imagePullSecrets.username` and `imagePullSecrets.password` on non-OpenShift Kubernetes can remain unaddressed for now.

I think, it is worth to also mention shut down of image registries hosted at `stackrox.io` and `collector.stackrox.io` addresses somewhere in release notes. https://srox.slack.com/archives/C3W3U0RGW/p1657047197148959
It's too late to include in the release but at least the online documentation will be clear about the shutdown.
